### ISSUE TITLE
Make RV_DONT_JUST_NULL_CHECK_READLINE can detect FileCacheImageOutputStream

### ISSUE THREAD

**Issue by Zustin** [spotbugs/spotbugs#1819]
Hi, I find a false negative about the rule [RV_DONT_JUST_NULL_CHECK_READLINE](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#rv-method-discards-result-of-readline-after-checking-if-it-is-non-null-rv-dont-just-null-check-readline). Please refer to the below code sample.

SpotBugs should have reported a violation at line 3, but it does not. Hence, I think this is a false negative. Thanks for your consideration😀.

```java
public void foo(OutputStream stream, File cacheDir) throws IOException {
    FileCacheImageOutputStream os = new FileCacheImageOutputStream(stream, cacheDir);
    while(os.readLine() != null) {}
}
```

MySpotBugs version is 4.4.2 and I run it by Idea plugin and CLI.

**Comment by gtoison** [comment]
This was fixed in #1823
